sklearn/ensemble/_weight_boosting.py

```diff
@@ -73,8 +73,9 @@ def __init__(self,
         self.random_state = random_state
 
     def _check_X(self, X):
-        return check_array(X, accept_sparse=['csr', 'csc'], ensure_2d=True,
-                           allow_nd=True, dtype=None)
+        return self._validate_data(
```

Comment:
I had to double check that `_check_X` was only called during non-`fit` methods. I think leaving a comment here would be best for now.

```py
def _check_X(self, X):
	# Only called to validate `X` in non-`fit` methods.
```